src/views/Layout.jsx

```diff
@@ -15,6 +15,23 @@ import {
 import logo from '../images/PredictaCartLogo.png';
 
 export function Layout() {
+	const location = useLocation();
+	let tabIndex;
+
+	switch (location.pathname) {
+		case '/':
+			tabIndex = 0;
+			break;
+		case '/list':
+			tabIndex = 1;
+			break;
+		case '/add-item':
+			tabIndex = 2;
+			break;
+		default:
+			tabIndex = 0;
+	}
+
 	return (
```

Comment:
- The `switch` is evaluating the value of `location.pathname`.
  
- `case '/':` checks if `location.pathname` equals `'/'`. If it does, the code within that case runs, setting `tabIndex` to `0`, and then the `break` statement ensures the switch statement stops executing.

- Similarly, `case '/list':` checks if `location.pathname` equals `'/list'`. If it does, the code within that case runs, setting `tabIndex` to `1`, and then breaks out of the switch.

- `case '/add-item':` does the same thing, checking against `'/add-item'` and potentially setting `tabIndex` to `2`.

- The `default` case is a catch-all. If none of the above cases match the value of `location.pathname`, the code inside the `default` case runs. Here, we're setting the `tabIndex` back to `0`.

The purpose of this logic is to determine which tab should be active based on the current URL path. By figuring out the active tab's index, we can then pass it to the Chakra `Tabs` component to programmatically highlight the correct tab. This ensures that when a user navigates using the logo (or any other means), the tab highlighting remains consistent with the current view.